### PR TITLE
Update mkchain

### PR DESCRIPTION
**Pull request by harryttd**

### CODE REVIEW

File: mkchain/tqchain/mkchain.py [hunk 1]
```diff
@@ -60,26 +68,39 @@ def get_genesis_vanity_chain_id(docker_image, seed_len=16):
...
     "zerotier_network": {"help": "Zerotier network id for external chain access"},
     "zerotier_token": {"help": "Zerotier token for external chain access"},
-    "bootstrap_peer": {"help": "peer ip to join"},
-    "docker_image": {
+    "expected_proof_of_work": {
+        "help": "Node identity generation difficulty",
+        "default": 0,
+        "type": int,
+    },
+    "bootstrap_peers": {
```
Comment (harryttd) [inline mkchain/tqchain/mkchain.py:98]: We should likely validate addresses. Old code converted them to ipv4 addresses. https://github.com/tqtezos/tezos-k8s/blame/d0092f19c690f815404c9d156cb9c1fe261284e5/mkchain.py#L281

**Comment by nicolasochem** [comment]
It looks like parameters file is uncommitted. Also, it does not work without any parameters:

```
[venv] nochem@peck ~/workspace/tezos-k8s () $ mkchain nico
Generating vanity chain id
Generating keys for account baker0
Traceback (most recent call last):
  File "/home/nochem/workspace/tezos-k8s/mkchain/venv/bin/mkchain", line 33, in <module>
    sys.exit(load_entry_point('mkchain', 'console_scripts', 'mkchain')())
  File "/home/nochem/workspace/tezos-k8s/mkchain/tqchain/mkchain.py", line 281, in main
    first_regular_node_name = next(iter(regular_nodes))
StopIteration
[venv] nochem@peck ~/workspace/tezos-k8s () $ mkchain nico --number-of-nodes 1
Generating vanity chain id
Generating keys for account baker0
Traceback (most recent call last):
  File "/home/nochem/workspace/tezos-k8s/mkchain/venv/bin/mkchain", line 33, in <module>
    sys.exit(load_entry_point('mkchain', 'console_scripts', 'mkchain')())
  File "/home/nochem/workspace/tezos-k8s/mkchain/tqchain/mkchain.py", line 292, in main
    with open(
FileNotFoundError: [Errno 2] No such file or directory: '/home/nochem/workspace/tezos-k8s/mkchain/tqchain/parameters.yaml'
```

**Comment by nicolasochem** [comment]
`mkchain nico` with zero parameters still fails. `mkchain nico --number-of-nodes 1` works fine and the resulting chain is able to boot up and start producing blocks.

I tried a zerotier chain, create worked but for invite I got this error:

```
│ Importing keys                                                                                                                                                                                                                                             │
│   Writing /var/tezos/client/secret_keys                                                                                                                                                                                                                    │
│   Writing /var/tezos/client/public_keys                                                                                                                                                                                                                    │
│   Writing /var/tezos/client/public_key_hashs                                                                                                                                                                                                               │
│ Starting config.json file generation                                                                                                                                                                                                                       │
│ Traceback (most recent call last):                                                                                                                                                                                                                         │
│   File "/entrypoint.py", line 490, in <module>                                                                                                                                                                                                             │
│     main()                                                                                                                                                                                                                                                 │
│   File "/entrypoint.py", line 115, in main                                                                                                                                                                                                                 │
│     create_node_config_json(                                                                                                                                                                                                                               │
│   File "/entrypoint.py", line 482, in create_node_config_json                                                                                                                                                                                              │
│     "values": {"genesis_pubkey": get_genesis_pubkey()},                                                                                                                                                                                                    │
│   File "/entrypoint.py", line 445, in get_genesis_pubkey                                                                                                                                                                                                   │
│     raise Exception("ERROR: Couldn't find the genesis_pubkey")                                                                                                                                                                                             │
│ Exception: ERROR: Couldn't find the genesis_pubkey                                                                                                                                                                                                         │
│ stream closed                                                 
```

**Comment by nicolasochem** [comment]
My previous issues are fixed.
I tried with 9.0-rc1 branch, the following worked:
* private chain with default params (just one baker)
* zerotier create/invite
* florencenet from rolling snapshot
* florencenet from full snapshot
* `mkchain nico --expected-proof-of-work 7 --number-of-nodes 1 --number-of-bakers 3`

It all works.

**Review by nicolasochem — APPROVED** [review]

**Pull request merged**